SpringProject/src/main/java/com/gradproject2019/users/repository/UserRepository.java

```diff
@@ -14,9 +14,20 @@
 public interface UserRepository extends JpaRepository<User, Long> {
 
     @Transactional
-    @Modifying
-    @Query("UPDATE User u SET u.firstName = IfNull(:firstName, u.firstName), u.lastName = IfNull(:lastName, u.lastName), u.username = IfNull(:username, u.username), u.email = IfNull(:email, u.email), u.occupation = IfNull(:occupation, u.occupation) WHERE u.id = :id")
-    void updateUser(@Param("id") Long id, @Param("firstName") String firstName, @Param("lastName") String lastName, @Param("username") String username, @Param("email") String email, @Param("occupation") String occupation);
+    @Modifying(clearAutomatically = true)
+    @Query("UPDATE User u SET " +
+            "u.firstName = IfNull(:firstName, u.firstName), " +
```

Comment:
you don't need table aliasing here (or in the other one). It doesn't hurt to have it but it doesn't look right

Reply:
I'm assuming you are referring to the `User u` and `u.firstName` bits?

Reply:
correcto